File: IT_support _App/IT_support_App/src/main/java/com/example/IT_support/_App/entities/Equipement.java
package com.example.IT_support._App.entities;

import com.example.IT_support._App.enums.EquipementStatus;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.List;

@Entity
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class Equipement {
        @Id
        @GeneratedValue(strategy = GenerationType.IDENTITY)
        private Long id;
        private String nome;
        private String type;
        @Enumerated(EnumType.STRING)
        private EquipementStatus staus;


       @OneToMany(mappedBy = "equipment", cascade = CascadeType.ALL, orphanRemoval = true)
        private List<Panne> panns;

}

File: IT-frontend/src/app/models/equipement.ts
import {Equipementstatus} from "../ennums/equipementstatus";
import {Panne} from "./panne";

export class Equipement {
  id: number;
  nome: string;
  type: string;
  status: Equipementstatus; // Corrected from 'staus' to 'status'
  panns: Panne[];

  constructor(id: number, nome: string, type: string, status: Equipementstatus, panns: Panne[]) {
    this.id = id;
    this.nome = nome;
    this.type = type;
    this.status = status; // Corrected here as well
    this.panns = panns;
  }

}
